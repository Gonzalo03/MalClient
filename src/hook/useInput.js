import { useState } from "react"

export const useInput = () => {

    const [input, setInput] = useState('');

    const handleInput = (e)=>{

        setInput(e.target.value)

    }

    const resetInput = () => {


        setInput('');

    }

   



    return [input, handleInput, resetInput]

}
