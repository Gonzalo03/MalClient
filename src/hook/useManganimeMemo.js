import { useMemo } from "react";


export const useManganimeMemo = (avgManganime) => {

      const animeMemo = useMemo(() => {
        const avg = avgManganime.reduce((a, b) => a + b, 0);
        return (avg / avgManganime.length).toString();
      }, [avgManganime]);
    
    return animeMemo.substr(0, animeMemo.indexOf('.') + 2);
}