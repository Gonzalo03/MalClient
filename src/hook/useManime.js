import { useEffect, useState } from "react"
import { request } from "../helpers/request"

export const useManime = (animeTag, context) => {

    const [manime, setManime] = useState({
        loading : true,
        data : null, 
        err : null
    })

    useEffect(
        ()=>request(context, animeTag)
        .then(data=>setManime({
            loading:false,
            data,
            err:null
        }))
        .catch(err=>setManime({
            loading:false,
            data:null,
            err
        }))
    , [animeTag, context])

    return manime;
}
