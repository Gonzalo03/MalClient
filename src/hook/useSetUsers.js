import { useState } from "react";

export const useSetUsers = () => {

    const [users, setUsers] = useState([]);

    const addUser = (tag) => {
    
        setUsers([...users, tag]);
    
      };

    const deleteUser = (tag) => {

        console.log("epaaw")
        
        return setUsers(
            users.filter(user=>user!==tag)
        )

    }
    
    return {users, addUser, deleteUser};

}
