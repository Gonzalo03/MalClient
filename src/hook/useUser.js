import { useEffect, useState } from "react"
import { request } from "../helpers/request"

export const useUser = (tag) => {
    
    const [user, setUser] = useState({
        loading:true,
        data:null,
        err:null
    })
    
    useEffect(
        ()=>request('user', tag)
        .then(data=>setUser({
                loading:false,
                data,
                err:null
            }))
        .catch(err=>setUser({
                loading:false,
                data:null,
                err
            }))
    , [tag])

        return user;

}
