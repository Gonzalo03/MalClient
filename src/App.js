import { UserDisplay } from "./components/UserDisplay";
import { useInput } from "./hook/useInput";
import { useSetUsers } from "./hook/useSetUsers";

import "./index.css";
const App = () => {
  
  const { users, addUser, deleteUser } = useSetUsers();

  const [input, handleInput, resetInput] = useInput();

  const handleSubmit = (e) => {

    e.preventDefault();

    addUser(input);

    resetInput();
  };

  return (
    <div className="container mx-auto">
      <div className="grid grid-cols-1 text-center">
        <div className="bg-primary text-custom-base mt-8 md:shadow">

          <h1 className="text-6xl font-yal">Mal Match</h1>

          <p className="text-3xl mt-5 font-sch">
            Demuestra quien es GOD y quien es ZzzzzZ
          </p>
        </div>

        <div className="w-full max-w-xs mx-auto mt-10">
          <form className="bg-white shadow-md rounded px-8 pt-6 pb-8 mb-4" onSubmit={handleSubmit}>
            <div className="mb-4">
              <label
                className="block text-custom-second text-sm font-bold mb-2"
                htmlFor ="usuario"
              >
                Usuario
              </label>
              <input
                className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
                id="usuario"
                name="usuario"
                onChange={handleInput}
                value={input}
                type="text"
                placeholder="Usuario"
              />
            </div>
            <div className="flex items-center justify-between">
              <button
                className="bg-primary hover:bg-secondary text-white font-bold py-2 px-4 rounded focus:outline-none focus:shadow-outline mx-auto"
                type="submit"
              >
                Buscar
              </button>
            </div>
          </form>
        </div>
      </div>

      <div className="grid grid-row-2 flex mt-5">

        {users[0] && <UserDisplay tag={users[0]} deleteUser={deleteUser} />}

        

        {users[1] && <UserDisplay tag={users[1]} deleteUser={deleteUser} />}
        

      </div>
      
    </div>
  );
};

export default App;
