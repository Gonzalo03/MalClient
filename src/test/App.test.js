import { mount } from "enzyme"
import App from "../App"
import { useSetUsers } from "../hook/useSetUsers";
import { userFix } from "./___fixtures___/userFix";
jest.mock("../hook/useSetUsers")

describe('Pruebas generales en el archivo App', () => {

    window.HTMLElement.prototype.scrollIntoView = function() {};

    /** useSetUsers */

    const deleteUser = jest.fn();
    const addUser = jest.fn();
    const users = [userFix,userFix];

    useSetUsers.mockReturnValue({
        users,
        addUser,
        deleteUser
    });
    
    const wrapper = mount(<App />)

    test('Debe hacer match con el snapshot', () => {
        

        expect(wrapper).toMatchSnapshot();
        
    })
    
    test('Debe de tener la cantidad de Usuarios en pantalla', () => {
        
        expect(wrapper.find('UserDisplay').length).toBe(users.length);
    })
    

    test('Debe de llamar al onSubmit del formulario', () => {
        
        const formSubmit = wrapper.find('form').prop('onSubmit');

        formSubmit({preventDefault(){}})

    })
    

})
