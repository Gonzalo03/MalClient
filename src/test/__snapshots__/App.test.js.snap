// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas generales en el archivo App Debe hacer match con el snapshot 1`] = `
<div
  className="container mx-auto"
>
  <div
    className="grid grid-cols-1 text-center"
  >
    <div
      className="bg-primary text-custom-base mt-8 md:shadow"
    >
      <h1
        className="text-6xl font-yal"
      >
        Mal Match
      </h1>
      <p
        className="text-3xl mt-5 font-sch"
      >
        Demuestra quien es GOD y quien es ZzzzzZ
      </p>
    </div>
    <div
      className="w-full max-w-xs mx-auto mt-10"
    >
      <form
        className="bg-white shadow-md rounded px-8 pt-6 pb-8 mb-4"
        onSubmit={[Function]}
      >
        <div
          className="mb-4"
        >
          <label
            className="block text-custom-second text-sm font-bold mb-2"
            htmlFor="usuario"
          >
            Usuario
          </label>
          <input
            className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
            id="usuario"
            name="usuario"
            onChange={[Function]}
            placeholder="Usuario"
            type="text"
            value=""
          />
        </div>
        <div
          className="flex items-center justify-between"
        >
          <button
            className="bg-primary hover:bg-secondary text-white font-bold py-2 px-4 rounded focus:outline-none focus:shadow-outline mx-auto"
            type="submit"
          >
            Buscar
          </button>
        </div>
      </form>
    </div>
  </div>
  <div
    className="grid grid-row-2 flex mt-5"
  >
    <div>
      <div
        className="sc-jTzLTM eAfuiW"
        style={
          Object {
            "margin": "auto",
            "positionAbsolute": "none",
          }
        }
      >
        <span
          className="sc-fjdhpX sc-jzJRlG ePZPHh"
          color="#536DFE"
          size="default"
        />
        <span
          className="sc-fjdhpX sc-cSHVUG eiUxHf"
          color="#536DFE"
          size="default"
        />
        <span
          className="sc-fjdhpX sc-kAzzGY iUEApe"
          color="#536DFE"
          size="default"
        />
        <span
          className="sc-fjdhpX sc-chPdSV bctrin"
          color="#536DFE"
          size="default"
        />
      </div>
    </div>
    <div>
      <div
        className="sc-jTzLTM eAfuiW"
        style={
          Object {
            "margin": "auto",
            "positionAbsolute": "none",
          }
        }
      >
        <span
          className="sc-fjdhpX sc-jzJRlG ePZPHh"
          color="#536DFE"
          size="default"
        />
        <span
          className="sc-fjdhpX sc-cSHVUG eiUxHf"
          color="#536DFE"
          size="default"
        />
        <span
          className="sc-fjdhpX sc-kAzzGY iUEApe"
          color="#536DFE"
          size="default"
        />
        <span
          className="sc-fjdhpX sc-chPdSV bctrin"
          color="#536DFE"
          size="default"
        />
      </div>
    </div>
  </div>
</div>
`;
